indexing-service/src/main/java/io/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -244,7 +297,9 @@ public TaskStatus call()
                                 command.add(nodeType);
                               }
 
-                              jsonMapper.writeValue(taskFile, task);
+                              if (!taskFile.exists()) {
```

Comment:
If it's totally corrupted (bad JSON), it will fail to restore. I can add a check for the task id.